bicycleparameters/cycle_app.py

```diff
@@ -213,6 +221,30 @@
 
 # creates generic set of Benchmark parameters
 
+@app.callback(Output('download-button', 'href'),
+              [Input('bike-dropdown', 'value'),
+               Input('range-slider', 'value')])
+
+def update_download_link(value,slider):
+    bicycle = bp.Bicycle(value, pathToData=path_to_app_data)
+    ctx = dash.callback_context
+    rangeSliderData = ctx.inputs.get('range-slider.value')
+    minBound = rangeSliderData[0]
+    maxBound = rangeSliderData[1]
+    steps = maxBound-minBound
+    speeds = np.linspace(minBound, maxBound, num=int(steps))
+    w,v = bicycle.eig(speeds)
+    result = np.dstack(
+        np.apply_along_axis(
+            lambda x: [x.real, x.imag], 0, w)
+        ).flatten().reshape(len(w),8)
+    speeds.shape = (len(speeds),1)
+    result = np.hstack((speeds,result))
+    result = pd.DataFrame(result)
+    result = result.set_axis(['speed','eval1_real','eval2_real','eval3_real','eval4_real','eval1_imag','eval2_imag','eval3_imag','eval4_imag'],axis = 1, inplace = False)
+    csv_string = result.to_csv(index=False, encoding='utf-8')
```

Comment:
You need to specify formatters here so that 14 decimal places are properly printed.